src/pages/recipes/index.md

```diff
@@ -451,3 +451,394 @@ function main(){
+
+```js
+var myEventListener = app.addEventListener("afterNew", myDisplayEventType);
+```
+The preceding script fragment refers to the following function:
+
+```js
+function myDisplayEventType(myEvent){
+alert("This event is the " + myEvent.eventType + " event.");
+}
+```
+
+**Removing an eventListener**
+
+To remove the eventListener created by the preceding script, run the following script:
+
+```js
+var myResult = app.removeEventListener("afterNew", myDisplayEventType);
+```
+When an eventListener responds to an event, the event may still be processed by other eventListeners that might be monitoring the event (depending on the propagation of the event). For example, the afterOpen event can be observed by eventListeners associated with both the application and the document.
+
+eventListeners do not persist beyond the current InDesign session. To make an eventListener available in every InDesign session, add the script to the startup scripts folder. 
+
+An event can trigger multiple eventListeners as it propagates through the scripting object model. The following sample script demonstrates an event triggering eventListeners registered to different objects:
+
+```js
+var myApplicationEventListener = app.eventListeners.add("beforeImport", myEventInfo);
+var myDocumentEventListener = app.documents.item(0).eventListeners.add("beforeImport", myEventInfo);
+function myEventInfo(myEvent){
+var myString = "Current Target: " + myEvent.currentTarget.name;
+alert(myString);
+}
+```
+When you run the preceding script and place a file, InDesign displays following information in sequence, and as alerts:
+1. Document name
+2. Application name
+
+Use the following script to remove the event listeners added by the preceding script:
+
+```js
+app.removeEventListener("beforeImport", myEventInfo);
+app.documents.item(0).removeEventListener("beforeImport", myEventInfo);
+```
+
+### Writing Events Script to Run Through Out the Session
+After the event listener is associated, the script is blocked by a promise and wait for the event in the background until the event listener is removed. If the event occurs, there is an automatic callback to execute the related function. If the event listener is removed, the promise can be resolved to complete the script execution.
+
+Here is a sample events script to run and listen for “afterNew” event throughout the InDesign session and generate an alert whenever a new document is created.
+
+```js
+let myInDesign = require("indesign");
+let app = myInDesign.app;
+ 
+await main();
+ 
+async function main()
+{
+  return new Promise((resolve, reject) => { mySnippet(); })
+}
+ 
+function mySnippet()
+{
+    //![Add event listener.]
+    var myEventListener = app.addEventListener("afterNew", myDisplayEventType);
+    //![Add event listener.]
+}
+ 
+//![Add event listener - functions.]
+function myDisplayEventType(myEvent)
+{
+    alert("This event is the " + myEvent.eventType + " event.");
```

Comment:
Why not replace this with a simple console.log?

Reply:
console.log() will output in the uxp error logs. The user might not be aware of the location of these logs. Using alert() we can simply see the output in a UI box.

Reply:
I think thats ok and its safe to assume that the logs in UDT would have the console.log statements. The code snippets should focus on the main topic with minimal noise.

Reply:
Agreed